debuglog-plugin/src/test/kotlin/com/bnorm/debug/log/IrPluginTest.kt

```diff
@@ -53,30 +65,146 @@ fun doSomething() {
+                throw t;
+            }
+        }
+    """.trimIndent(),
 
-    val out = invokeMain(result, "MainKt").trim().split("""\r?\n+""".toRegex())
-    assert(out.size == 8)
-    assert(out[0] == "⇢ greet(greeting=Hello, name=World)")
-    assert(out[1].matches("⇠ greet \\[\\d+(\\.\\d+)?ms] = Hello, World!".toRegex()))
-    assert(out[2] == "Hello, World!")
-    assert(out[3] == "⇢ greet(greeting=Hello, name=Kotlin IR)")
-    assert(out[4].matches("⇠ greet \\[\\d+(\\.\\d+)?ms] = Hello, Kotlin IR!".toRegex()))
-    assert(out[5] == "Hello, Kotlin IR!")
-    assert(out[6] == "⇢ doSomething()")
-    assert(out[7].matches("⇠ doSomething \\[\\d+(\\.\\d+)?ms]".toRegex()))
+      expectedDoSomethingMethod = """
+        public static final void doSomething() {
+            System.out.println((Object)"⇢ doSomething()");
+            final TimeMark markNow = TimeSource.Monotonic.INSTANCE.markNow();
+            try {
+                Thread.sleep(15L);
+                System.out.println((Object)("⇠ doSomething [" + (Object)Duration.toString-impl(markNow.elapsedNow-UwyO8pc()) + ']'));
+            }
+            catch (Throwable t) {
+                System.out.println((Object)("⇠ doSomething [" + (Object)Duration.toString-impl(markNow.elapsedNow-UwyO8pc()) + "] = " + t));
+                throw t;
+            }
+        }
+    """.trimIndent(),
+
+      outputVerifyFunc = { out ->
+        assert(out.size == 8)
+        assert(out[0] == "⇢ greet(greeting=Hello, name=World)")
+        assert(out[1].matches("⇠ greet \\[\\d+(\\.\\d+)?ms] = Hello, World!".toRegex()))
+        assert(out[2] == "Hello, World!")
+        assert(out[3] == "⇢ greet(greeting=Hello, name=Kotlin IR)")
+        assert(out[4].matches("⇠ greet \\[\\d+(\\.\\d+)?ms] = Hello, Kotlin IR!".toRegex()))
+        assert(out[5] == "Hello, Kotlin IR!")
+        assert(out[6] == "⇢ doSomething()")
+        assert(out[7].matches("⇠ doSomething \\[\\d+(\\.\\d+)?ms]".toRegex()))
+      }
+    )
   }
 
   @Test
   fun `IR plugin disabled`() {
-    val result = compile(sourceFile = main, DebugLogComponentRegistrar(false))
+    testIrPlugin(
+      isPluginEnabled = false,
+      expectedGreetMethod = """
+        public static final String greet(@NotNull final String greeting, @NotNull final String name) {
+            Intrinsics.checkNotNullParameter(greeting, "greeting");
+            Intrinsics.checkNotNullParameter(name, "name");
+            if (!Intrinsics.areEqual(greeting, "Hello")) {
+                return "Early return";
+            }
+            Thread.sleep(15L);
+            return greeting + ", " + name + '!';
+        }
+      """.trimIndent(),
+
+      expectedDoSomethingMethod = """
+        public static final void doSomething() {
+            Thread.sleep(15L);
+        }
+    """.trimIndent(),
+
+      outputVerifyFunc = { out ->
+        assertTrue(out.size == 2)
+        assertEquals("Hello, World!", out[0])
+        assertEquals("Hello, Kotlin IR!", out[1])
```

Comment:
For what it's worth, I used `assertTrue` so I could dog food my [kotlin-power-assert](https://github.com/bnorm/kotlin-power-assert) plugin. I would prefer to keep it this way. Just something to keep in mind as more assertion statements are added.

Reply:
How odd. I don't recall changing that. I will revert that